### PR TITLE
sys/event/timeout: add timeout_clear() function

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
For some use-cases, it is required to cancel scheduled events (e.g. to model timeouts). This PR adds a `event_timeout_clear()` function to the `event/timout` submodule to enable this.

### Issues/PRs references
none

### CODE REVIEW

File: sys/include/event/timeout.h [hunk 1]
```diff
@@ -54,11 +54,12 @@ typedef struct {
 /**
  * @brief   Initialize timeout event object
  *
- * @param[in]   event_timeout   event_timeout object to initilalize
+ * @param[in]   event_timeout   event_timeout object to initialize
```
Comment (miri64) [inline sys/include/event/timeout.h]: Unrelated, but okay if separate commit.

File: sys/include/event/timeout.h [hunk 2]
```diff
@@ -54,11 +54,12 @@ typedef struct {
...
  * @param[in]   queue           queue that the timed-out event will be added to
  * @param[in]   event           event to add to queue after timeout
  */
-void event_timeout_init(event_timeout_t *event_timeout, event_queue_t *queue, event_t *event);
+void event_timeout_init(event_timeout_t *event_timeout, event_queue_t *queue,
+                        event_t *event);
```
Comment (miri64) [inline sys/include/event/timeout.h]: Dito

File: sys/include/event/timeout.h [hunk 3]
```diff
@@ -69,11 +70,23 @@ void event_timeout_init(event_timeout_t *event_timeout, event_queue_t *queue, ev
  * @note: the used event_timeout struct must stay valid until after the timeout
  *        event has been processed!
  *
- * @param[in]   event_timeout   event_timout context onject to use
+ * @param[in]   event_timeout   event_timout context object to use
```
Comment (miri64) [inline sys/include/event/timeout.h]: Dito

File: tests/events/main.c [hunk 4]
```diff
@@ -69,7 +71,18 @@ static void timed_callback(void *arg)
     uint32_t now = xtimer_now_usec();
     assert((now - before >= 100000LU));
     printf("triggered timed callback with arg 0x%08x after %" PRIu32 "us\n", (unsigned)arg, now - before);
-    printf("[SUCCESS]\n");
+    puts("[SUCCESS]");
```
Comment (miri64) [inline tests/events/main.c]: Unrelated, but okay, if provided in a separate PR.

File: tests/events/main.c [hunk 5]
```diff
@@ -85,17 +98,25 @@ int main(void)
     printf("canceling 0x%08x\n", (unsigned)&event2);
     event_cancel(&queue, &event2);
 
-    printf("posting custom event\n");
+    puts("posting custom event");
```
Comment (miri64) [inline tests/events/main.c]: Dito

File: tests/events/main.c [hunk 6]
```diff
@@ -85,17 +98,25 @@ int main(void)
...
     event_post(&queue, (event_t *)&custom_event);
 
     event_timeout_t event_timeout;
 
-    printf("posting timed callback with timeout 1sec\n");
+    puts("posting timed callback with timeout 1sec");
```
Comment (miri64) [inline tests/events/main.c]: Dito

File: tests/events/main.c [hunk 7]
```diff
@@ -85,17 +98,25 @@ int main(void)
...
     event_timeout_init(&event_timeout, &queue, (event_t *)&event_callback);
     before = xtimer_now_usec();
-    event_timeout_set(&event_timeout, 100000LU);
+    event_timeout_set(&event_timeout, (1 * US_PER_SEC));
```
Comment (miri64) [inline tests/events/main.c]: Dito

File: tests/events/main.c [hunk 8]
```diff
@@ -85,17 +98,25 @@ int main(void)
...
+    event_timeout_t event_timeout_canceled;
+
+    puts("posting timed callback with timeout 0.5sec and canceling it again");
+    event_timeout_init(&event_timeout_canceled, &queue,
+                       (event_t *)&noevent_callback);
+    event_timeout_set(&event_timeout_canceled, 500 * US_PER_MS);
+    event_timeout_clear(&event_timeout_canceled);
 
-    printf("launching event queue\n");
+    puts("launching event queue");
```
Comment (miri64) [inline tests/events/main.c]: Dito

**Comment by haukepetersen** [comment]
Split commits.

**Comment by miri64** [comment]
Thanks

**Comment by haukepetersen** [comment]
Thanks!

**Pull request merged**